crates/worker/src/partition/state_machine/lifecycle/pinned_deployment.rs

```diff
@@ -62,6 +84,18 @@ where
             .await
             .map_err(Error::Storage)?;
 
+        if should_apply_cancellation_hotfix {
+            trace!(
+                "Applying hotfix for cancellation when invocation doesn't have a pinned service protocol"
+            );
+            // TODO this is the code when we'll get rid of protocol <= 3
+            // OnCancelCommand {
+            //     invocation_id,
+            //     invocation_status,
+            // }.apply(ctx).await
```

Comment:
Once we remove protocol <= 3, then we also no longer need the `hotfix_apply_cancellation_after_deployment_is_pinned` field in the `InvocationStatus`, right? Because then we know it's the at least protocol 4 which requires us to forward the signal?